File: golang/src/p10/main.go
package main

import "fmt"
import "acommon"

func main() {
	sum := uint64(2)

	for i := 3; i < 2000000; i += 2 {
		if acommon.IsPrime(i) {
			sum += uint64(i)
		}
	}

	fmt.Println(sum)
}

File: rust/src/p10/main.rs
fn main() {
  let mut sum: u64 = 0;
  for i in 1..2000000 {
    if is_prime(i) {
      sum += i as u64
    }
  }

  println!("{}", sum);
}

fn is_prime(n: i32) -> bool {
  for i in 2..=n/2 {
    if n % i == 0 {
      return false;
    }
  }
  true
}
